packages/components/src/DateTimePickers/pickers/DatePicker/DatePicker.component.js

```diff
@@ -1,83 +1,166 @@
+	isSelectedDate(date) {
+		return this.props.selectedDate !== undefined
+			&& isSameDay(this.props.selectedDate, date);
 	}
 
-	return (
-		<div className={theme.container}>
-			<div className={classNames(theme['calendar-row'], theme['calendar-header-row'])}>
-				{dayNames.map((dayName, i) =>
-					<abbr
-						className={theme['calendar-item']}
-						key={i}
-						title={dayName}
-					>
-						{dayName.charAt(0)}
-					</abbr>
-				)}
-			</div>
+	isDisabledDate(dateToCheck) {
+		const disabledRules = this.props.disabledRules;
+		if (!disabledRules) {
+			return false;
+		}
 
-			<hr className={theme.separator} />
+		return disabledRules
+			.some(disabledRule => {
+				if (typeof disabledRules !== 'object') {
+					return false;
+				}
 
-			{weeks.map((week, i) =>
-				<div
-					className={classNames(theme['calendar-row'], theme['calendar-body-row'])}
-					key={i}
-				>
-					{week.map((day, j) =>
+				switch (disabledRule.constructor) {
+					case Date:
+						return isSameDay(disabledRule, dateToCheck);
+
+					case Array:
+						return disabledRule
+							.some(disabledDateRule => isSameDay(disabledDateRule, dateToCheck));
+
+					default:
+						return false;
+				}
+			});
+	}
+
+	isCurrentMonth(date) {
+		return getMonth(date) === this.props.calendar.monthIndex;
+	}
+
+	render() {
+		const { year, monthIndex } = this.props.calendar;
+
+		const weeks = this.getWeeks(year, monthIndex, FIRST_DAY_OF_WEEK);
+		const dayNames = getDayNames(FIRST_DAY_OF_WEEK);
+
+		return (
+			<div className={theme.container}>
+				<div className={theme['calendar-header']}>
+					<div className={classNames(theme['calendar-row'])}>
+						{dayNames.map((dayName, i) =>
+							<abbr
+								className={theme['calendar-item']}
+								key={i}
+								title={dayName}
+							>
+								{dayName.charAt(0)}
+							</abbr>
+						)}
+					</div>
+					<hr className={theme.separator} />
+				</div>
+
+
+				<div className={theme['calendar-body']}>
+					{weeks.map((week, i) =>
 						<div
-							className={theme['calendar-item']}
-							key={j}
+							className={classNames(theme['calendar-row'], theme['calendar-body-row'])}
+							key={i}
 						>
-							<DayPickerAction
-								label={day.number.toString()}
-								isSelectedDay={isSelectedDay(day.number)}
-								isDisabledDay={isDisabledDay(day.number)}
-								isCurrentDay={isCurrentDay(day.number)}
-								aria-label={isDisabledDay(day.number)
-									? 'Unselectable date'
-									: `Select '${day.number}'`}
-							/>
+							{week.map((date, j) =>
+								<div
+									className={theme['calendar-item']}
+									key={j}
+								>
+									{
+										this.isCurrentMonth(date) &&
+										<DayPickerAction
+											label={getDate(date).toString()}
+											isSelected={this.isSelectedDate(date)}
+											isDisabled={this.isDisabledDate(date)}
+											isToday={isToday(date)}
+											aria-label={this.isDisabledDate(date)
+												? 'Unselectable date'
```

Comment:
Remove that since you don't render it if it is disabled.
And it avoid testing the disable status twice

Reply:
The action/button is always render, just `disabled`.
But i can memoize or put a variable for re-usability or just have the same aria-label but it seems misleading ❓

Reply:
Oh you're right, I misread this code.
Yes please, check it once and use the result twice

Reply:
✔